### ISSUE TITLE
`is_holiday` function doesn't return correct result when pandas `Timestamp` is passed. 

### ISSUE THREAD

**Issue by yassineAlouini** [peopledoc/workalendar#294]
The `is_holiday` function expects a `datetime.datetime` format. However, it doesn't raise any error when a pandas `Timestamp` is passed (https://pandas.pydata.org/pandas-docs/version/0.23.4/generated/pandas.Timestamp.html). 

Here is a code example (`workalendar` version '3.1.1'):  

```
from workalendar import europe
import pandas as pd
import datetime


fr = europe.France()
# Should be True and returns True
print(fr.is_holiday(datetime.datetime(2017, 1, 1)))
# Should be True but returns False
print(fr.is_holiday(pd.to_datetime("2017-1-1")))
# To fix this, use the `to_pydatetime` method
print(fr.is_holiday(pd.to_datetime("2017-1-1").to_pydatetime()))

```
I expect the library to raise an error when the passed input isn't of type `datetime.datetime`.
What are your thoughts? 
Thanks in advance. :)

**Comment by yassineAlouini** [comment]
@brunobord Yes, it is using the `.to_pydatetime()` method: 
```
import pandas as pd
tms = pd.to_datetime('2017-1-1')
print(type(tms))
# pandas._libs.tslibs.timestamps.Timestamp
print(type(tms.to_pydatetime()))
# datetime.datetime
```

**Comment by brunobord** [comment]
ok, looks like it's doable.

Then my question would be: why should `workalendar` accept pandas Timestamps? I've had an afterthought and I'm not sure workalendar would support exotic date/datetime input types.
If we accept pandas timestamps, why not accepting strings like ``YYYY-MM-DD``, and then integers (as a unix timestamp), and then a picture of a calendar as an input? (yes, I'm exaggerating, for the sake of the argument).
I'm not saying that I want to *reject* this possibility, I'm saying that I think I'd need a serious argument to accept non-standard types in our API.

**Comment by yassineAlouini** [comment]
I see your point. 

I am not advising to accept additional types. Maybe a check of type + raising an exception would be enough to avoid dealing with other formats than `datetime.datetime`? 

What do you think?

**Comment by johnberroa** [comment]
I had the same problem.  Throwing an error would be nice because if I didn't double check I would have never known.

**Comment by brunobord** [comment]
turns out pandas' Timeout **is** a datetime subclass. Interesting, we may be able to support more types, as long as they are something like "duck dates" (that quack like a date)

**Comment by brunobord** [comment]
closed by #298 + released with [the version 3.2.0](https://github.com/peopledoc/workalendar/releases/tag/3.2.0)
